cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -1189,40 +1191,55 @@ TEST_F(TestS3FS, OpenInputFile) {
   ASSERT_RAISES(IOError, file->Seek(10));
 }
 
-TEST_F(TestS3FS, OpenOutputStreamBackgroundWrites) { TestOpenOutputStream(); }
+struct S3OptionsTestParameters {
+  bool background_writes{false};
+  bool allow_delayed_open{false};
 
-TEST_F(TestS3FS, OpenOutputStreamSyncWrites) {
-  options_.background_writes = false;
-  MakeFileSystem();
-  TestOpenOutputStream();
-}
-
-TEST_F(TestS3FS, OpenOutputStreamAbortBackgroundWrites) { TestOpenOutputStreamAbort(); }
+  void apply_to_s3_options(S3Options& options) const {
+    options.background_writes = background_writes;
+    options.allow_delayed_open = allow_delayed_open;
+  }
```

Comment:
Coding style nit: 1) use `CamelCase` for function names, 2) avoid mutable refs; therefore:
```suggestion
  void ApplyToS3Options(S3Options* options) const {
    options->background_writes = background_writes;
    options->allow_delayed_open = allow_delayed_open;
  }
```